### PR TITLE
Feature/transactions and batch inserts

### PR DESCRIPTION
**Pull request by rmothilal**
exposing knex database object to dependencies to allow access to multiple tables, transactions and batches. All functionality that was previously blocked

### CODE REVIEW

**Comment by mdebarros** [comment]
Please link to a story!

File: test/unit/database.test.js [hunk 1]
```diff
@@ -15,21 +15,21 @@ Test('database', databaseTest => {
...
+  let tableNames = [{TABLE_NAME: 'accounts'}, {TABLE_NAME: 'users'}, {TABLE_NAME: 'tokens'}]
 
   databaseTest.beforeEach(t => {
     sandbox = Sinon.sandbox.create()
 
     knexConnStub = sandbox.stub()
     knexConnStub.destroy = sandbox.stub()
-    knexConnStub.client = { config: { client: 'mysql' } }
-    knexConnStub.withArgs('information_schema.tables').returns({ where: sandbox.stub().withArgs('TABLE_SCHEMA', 'databaseSchema').returns({ select: sandbox.stub().withArgs('TABLE_NAME').returns(P.resolve(tableNames)) }) })
+    knexConnStub.client = {config: {client: 'mysql'}}
```
Comment (mdebarros) [inline test/unit/database.test.js]: Is there anyway we can make this config?

Comment (rmothilal) [reply test/unit/database.test.js]: make this config for the test? I'm confused as to why you would do that?

Comment (rmothilal) [reply test/unit/database.test.js]: why would there need to be config for the test

Comment (mdebarros) [reply test/unit/database.test.js]: Ah, didnt see it was for a test.

Never-mind :D

**Comment by mdebarros** [comment]
Will approve once a story is linked.

**Comment by rmothilal** [comment]
@mdebarros Story is linked

**Pull request closed without merging**